core/src/main/scala/org/apache/spark/h2o/converters/SparkDataFrameConverter.scala

```diff
@@ -144,14 +153,13 @@ trait SparkDataFrameConverter extends Logging with ConverterUtils {
         }
 
       }
-      con.increaseRowCounter()
     })
 
     //Compress & write data in partitions to H2O Chunks
     con.closeChunks()
 
     // Return Partition number and number of rows in this partition
-    (context.partitionId, con.numOfRows)
+    (context.partitionId, iterator.length)
```

Comment:
The same as above:
  - the iterator.length/iterator.size should not be used - instead of that pls compute number of items in foreach call.